rules/no-error-ctor-with-notthrows.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+const {visitIf} = require('enhance-visitors');
+const util = require('../util');
+const createAvaRule = require('../create-ava-rule');
+
+const errorNameRegex = /^([A-Z][a-z\d]*)*Error$/;
+
+const create = context => {
+	const ava = createAvaRule();
+
+	return ava.merge({
+		CallExpression: visitIf([
+			ava.isInTestFile,
+			ava.isInTestNode
+		])(node => {
+
+			if (typeof node.callee.property === 'undefined') {
+				return;
+			}
+
+			const functionArgIndex = node.arguments.length - 1;
+
+			if (functionArgIndex !== 1) {
+				return;
+			}
+
+			const calleeProperty = node.callee.property.name;
+			const functionArgName = node.arguments[1].name;
```

Comment:
We should do nothing when `node.arguments.length < 2`, there will be other rules take care of the `t.notThrows` signature.

Also, we should check if `node.arguments[1]` is an `Identifier`, hence we retrieve the identifier name without ambiguity.

Reply:
Isn't that case covered under
```
const functionArgIndex = node.arguments.length - 1;
// Here we are ignoring all other cases which dont have two arguments.
if (functionArgIndex !== 1) {
	return;
}
```
I'm curious, Can you give a failing case for ambiguity.

Reply:
> const functionArgIndex = node.arguments.length - 1;

Sorry I miss this line. It should be good to go.

> Can you give a failing case for ambiguity.

Sure, hence `node.arguments[1]` might not be an `Identifier` in some cases, e.g.
```
t.notThrows(() => {
        t.pass();
    }, void 0);
```
`node.arguments[1]` is a `UnaryExpression`, this node does not have a `.name` property. Instead of throwing error, I think we should skip it as it should be handled by other rules.

Reply:
Yeah, I agree with this, adding it to the upcoming commit. Thanks.